smithy-model/src/main/java/software/amazon/smithy/model/knowledge/IdentifierBindingIndex.java

```diff
@@ -87,22 +93,73 @@ public BindingType getOperationBindingType(ToShapeId resource, ToShapeId operati
      * @return Returns the identifier bindings map or an empty map if the
      *  binding is invalid or cannot be found.
      */
-    public Map<String, String> getOperationBindings(ToShapeId resource, ToShapeId operation) {
-        return Optional.ofNullable(bindings.get(resource.toShapeId()))
+    public Map<String, String> getOperationInputBindings(ToShapeId resource, ToShapeId operation) {
+        return Optional.ofNullable(inputBindings.get(resource.toShapeId()))
+                .flatMap(resourceMap -> Optional.ofNullable(resourceMap.get(operation.toShapeId())))
+                .map(Collections::unmodifiableMap)
+                .orElseGet(Collections::emptyMap);
+    }
+
+    /**
+     * Gets a map of identifier names to output member names that provide a
+     * value for that identifier.
+     *
+     * @param resource Shape ID of a resource.
+     * @param operation Shape ID of an operation.
+     * @return Returns the identifier bindings map or an empty map if the
+     *  binding is invalid or cannot be found.
+     */
+    public Map<String, String> getOperationOutputBindings(ToShapeId resource, ToShapeId operation) {
+        return Optional.ofNullable(outputBindings.get(resource.toShapeId()))
                 .flatMap(resourceMap -> Optional.ofNullable(resourceMap.get(operation.toShapeId())))
                 .map(Collections::unmodifiableMap)
                 .orElseGet(Collections::emptyMap);
     }
 
+    /**
+     * Gets a map of identifier names to input member names that provide a
+     * value for that identifier.
+     *
+     * @deprecated Use {@link #getOperationInputBindings} instead.
+     *
+     * @param resource Shape ID of a resource.
+     * @param operation Shape ID of an operation.
+     * @return Returns the identifier bindings map or an empty map if the
+     *  binding is invalid or cannot be found.
+     */
+    @Deprecated
+    public Map<String, String> getOperationBindings(ToShapeId resource, ToShapeId operation) {
+        return getOperationInputBindings(resource, operation);
+    }
+
+    /**
+     * Gets a set of identifier names associated with either input or output on the operation.
+     *
+     * @param resource Shape ID of a resource.
+     * @param operation Shape ID of an operation.
+     * @return Returns a set of all identifier names associated with the operation.
+     */
+    public Set<String> getAllOperationIdentifiers(ToShapeId resource, ToShapeId operation) {
```

Comment:
Is this necessary? I haven't looked if it's used elsewhere. If not, I'd remove it.